usaspending_api/common/data_connectors/async_spending_by_award_count.py

```diff
@@ -0,0 +1,25 @@
+import asyncio
+
+from django.db.models import Count
+
+from usaspending_api.awards.v2.filters.matview_filters import matview_search_filter
+from usaspending_api.common.helpers.orm_helpers import generate_raw_quoted_query
+from usaspending_api.common.data_connectors.async_sql_query import async_run_select
+
+
+def async_fetch_category_counts(filters, category_to_model_dict):
```

Comment:
Nice! and good use of naming convention to demarcate an async function. Should this method be declared with `async def async_fetch_category_counts(...)`?
I haven't written code using this, but it may be recommended to use the `async`/`await` keywords for code like this: https://docs.python.org/3.5/library/asyncio-task.html

i.e. so that when this is called, it is _awaited_

e.g. 
```
def some_function()
    # ...
    await async_fetch_category_counts(filters, category_to_model_dict)
    # ...
```

If needed, it should be applied here: https://github.com/fedspendingtransparency/usaspending-api/pull/1977/files#r316344546

Reply:
That's what async_run_select() is in usaspending_api/common/data_connectors/async_spending_by_award_count.py.

Perhaps this function *shouldn't* say async since it calls the async function

Reply:
Yes - I think you're right that this function _is not_ async (shouldn't be _awaited_) and should drop that naming convention. 

Brings up the other question:
[`results[k] = asyncio.ensure_future(async_run_select(sql[:remove_groupby_string_index]), loop=loop)`](https://github.com/fedspendingtransparency/usaspending-api/pull/1977/files#diff-954252da19751382babb8e78569beafbR19)

Should ^^ code have `await`? Or is the `asyncio.ensure_future(...)` an alternative way to _await_ the async result?

Reply:
My understanding is
`asyncio.ensure_future()` creates the "promise" the task will be run in the future (like a queue)
`asyncio.gather()` will run all the Tasks _concurrently_
`run_until_complete()` is blocking until all Tasks complete and return

Reply:
I see... the example https://docs.python.org/3.5/library/asyncio-task.html#example-parallel-execution-of-tasks would seem to indicate that you don't necessarily need to call `asyncio.ensure_future(...)`, but could pass the collection of async callables (aka coroutines, that is the calls to `async_run_select`) directly into `asyncio.gather(...)`. 

But either way, it looks like the code will work. So all good.

Reply:
Good find. I tried several approaches to get the coroutines to A: run, B: block until all were done and C: obtain the results. I stopped once it worked without thinking of checking for extraneous code.